### PR TITLE
Review and update Developer Guide

### LINKED ISSUE #142: Update and Review DG
### Description

- [x] Update any remaining portions of the Developer Guide and remove any parts that are not relevant to Doctrack
- [x] Carry out final review of Developer Guide as a whole
- [x] Save the DG as a pdf file using this [procedure](https://se-education.org/guides/tutorials/savingPdf.html)

### PR DESCRIPTION
**Pull request by rxchell**
Fixes #142

### CODE REVIEW

**Review by zaidansani — APPROVED** [review]
LGTM - formatting/diagrams

**Comment by choiwab** [comment]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -450,16 +481,11 @@ When a user types a `help` command, the DocTrack application will display a `Hel
 **Aspect: Save patient and appointment data in:**
 <br>
 * **Alternative 1 (current choice):** two different files, patient data in `data/addressbook.json` and appointment data in `data/appointmentbook.json`.
-    * Pros: 
-        * More organised file management
-        * Quicker read and write times for each file
-    * Cons: 
-        * Higher chance of inconsistencies between patient and appointment data
+    * **Pros**: More organised file management, with a quicker read and write times for each file.
+  * **Cons**: Higher chance of inconsistencies between patient and appointment data.
```
Comment (jayjay19630) [inline docs/DeveloperGuide.md:485]: Is this line missing an indent?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -223,157 +235,179 @@ The activity diagram shows the general sequence of steps when a user interacts w
...
 
-**Aspect: Whether to implement case sensitivity in matching**
-
+**Aspect: Whether to implement case sensitivity in matching for search terms**
 - **Alternative 1 (Current choice)**: Implement case-insensitive matching for search terms.
-    - **Pros**: Enhances user experience by allowing searches to ignore case differences
-    - **Cons**: Slightly more processing required to normalize case during search
-
+    - **Pros**: Enhances user experience by allowing searches to ignore case differences.
+    - **Cons**: Slightly more processing required to normalize case .during search
```
Comment (jayjay19630) [inline docs/DeveloperGuide.md:401]: There is a slight errata here with the fullstop in the middle of the sentence. Please fix!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -488,96 +514,59 @@ is not specified, it would be represented as `"null"`, in the `appointmentbook.j
...
-        * More verbose and could result in information overload
-* **Alternative 2**: Show only one list at a time, but toggle between the two using a `list appt` or `list person` command
-    * Pros:
-        * Information is simpler to digest
-    * Cons:
-        * More overhead of handling switching between lists
-        * Difficult to cross reference when typing certain commands
+* **Alternative 1 (current choice):** Show lists as two separate panels side by side.
+  * **Pros**: Easier to see all information at once, and easier to cross-reference when doing `add appt` 
+    or `edit appt` commands, which may need information about `personId`.
```
Comment (jayjay19630) [inline docs/DeveloperGuide.md:554]: Thanks for making reference to personId consistent throughout entire DG

**Review by jayjay19630 — CHANGES_REQUESTED** [review]
LGTM for the design considerations, just some typos that need fixing.

**Review by jayjay19630 — APPROVED** [review]
LGTM

**Pull request merged**